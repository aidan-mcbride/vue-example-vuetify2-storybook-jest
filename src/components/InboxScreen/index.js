import InboxScreen from './InboxScreen.vue';
import BaseInboxScreen from './BaseInboxScreen.vue';

export { InboxScreen, BaseInboxScreen };

export default InboxScreen;
