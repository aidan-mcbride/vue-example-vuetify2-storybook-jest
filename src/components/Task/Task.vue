<template>
  <v-list-item>
    <v-list-item-action>
      <v-btn
        class="task__archive-button"
        aria-label="archive task"
        text
        icon
        @click="$emit('archiveTask', task.id)"
      >
        <v-icon>{{ checkboxIcon }}</v-icon>
      </v-btn>
    </v-list-item-action>
    <v-list-item-content>
      <v-list-item-title>{{ task.title }}</v-list-item-title>
    </v-list-item-content>
    <v-list-item-action>
      <v-btn
        v-if="!isChecked"
        class="task__pin-button"
        aria-label="pin task"
        text
        icon
        @click="$emit('pinTask', task.id)"
      >
        <v-icon v-if="isPinned" color="blue">mdi-star</v-icon>
        <v-icon v-else>mdi-star-outline</v-icon>
      </v-btn>
    </v-list-item-action>
  </v-list-item>
</template>

<script>
export default {
  name: 'Task',
  props: {
    task: {
      type: Object,
      required: true,
      default: () => ({
        id: '',
        state: '',
        title: ''
      })
    }
  },
  computed: {
    isChecked() {
      return this.task.state === 'TASK_ARCHIVED';
    },
    isPinned() {
      return this.task.state === 'TASK_PINNED';
    },
    checkboxIcon() {
      return this.isChecked
        ? 'mdi-checkbox-marked-outline'
        : 'mdi-checkbox-blank-outline';
    }
  }
};
</script>
