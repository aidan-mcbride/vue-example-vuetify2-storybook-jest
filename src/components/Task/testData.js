export const taskData = {
  id: '1',
  title: 'Test Task',
  state: 'TASK_INBOX',
  updated_at: new Date(2019, 0, 1, 9, 0)
};
