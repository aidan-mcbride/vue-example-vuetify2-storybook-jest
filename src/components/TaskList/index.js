import TaskList from './TaskList.vue';
import BaseTaskList from './BaseTaskList.vue';

export { TaskList, BaseTaskList };
