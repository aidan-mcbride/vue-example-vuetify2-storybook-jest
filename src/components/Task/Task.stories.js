import { action } from '@storybook/addon-actions';
import Task from './Task';

import { taskData } from './testData';
export default {
  title: 'Task',
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/
};
export const actionsData = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask')
};

const taskTemplate = `<task :task="task" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`;

// default task state
export const Default = () => ({
  components: { Task },
  template: taskTemplate,
  props: {
    task: {
      default: () => taskData
    }
  },
  methods: actionsData
});
// pinned task state
export const Pinned = () => ({
  components: { Task },
  template: taskTemplate,
  props: {
    task: {
      default: () => ({
        ...taskData,
        state: 'TASK_PINNED'
      })
    }
  },
  methods: actionsData
});
// archived task state
export const Archived = () => ({
  components: { Task },
  template: taskTemplate,
  props: {
    task: {
      default: {
        ...taskData,
        state: 'TASK_ARCHIVED'
      }
    }
  },
  methods: actionsData
});
