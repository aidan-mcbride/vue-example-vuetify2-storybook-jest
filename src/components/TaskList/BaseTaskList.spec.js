// Components
import BaseTaskList from './BaseTaskList';
import { Task } from '@/components/Task';

// Utilities
import { mount } from '@vue/test-utils';
import { defaultTasksData, withPinnedTasksData } from './testData';

describe('BaseTaskList', () => {
  const build = propsData => {
    const wrapper = mount(BaseTaskList, { propsData });

    return {
      wrapper
    };
  };

  describe('Default State', () => {
    it('renders correct markup', () => {
      const { wrapper } = build({ tasks: defaultTasksData });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('With Pinned Tasks', () => {
    it('renders correct markup', () => {
      const { wrapper } = build({ tasks: withPinnedTasksData });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders pinned tasks at the start of the list', () => {
      const { wrapper } = build({ tasks: withPinnedTasksData });
      const firstTaskPinned = wrapper.find(Task);

      expect(firstTaskPinned.props().task.state).toBe('TASK_PINNED');
    });
  });

  describe('Loading State', () => {
    it('renders correct markup', () => {
      const { wrapper } = build({ loading: true });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('Empty State', () => {
    it('renders correct markup', () => {
      const { wrapper } = build();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
