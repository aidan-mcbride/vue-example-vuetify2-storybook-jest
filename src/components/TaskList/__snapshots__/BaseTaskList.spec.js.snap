// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTaskList Default State renders correct markup 1`] = `
<div>
  <!---->
  <!---->
  <div role="list" class="v-list v-sheet v-sheet--tile theme--light">
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 1</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 2</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 3</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 4</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 5</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 6</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;

exports[`BaseTaskList Empty State renders correct markup 1`] = `
<div>
  <!---->
  <div role="alert" class="v-alert v-sheet theme--dark v-alert--prominent success">
    <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon material-icons theme--dark">$success</i>
      <div class="v-alert__content">
        <div class="title">You have no tasks</div>
        <div>Sit back and relax</div>
      </div>
    </div>
  </div>
  <!---->
</div>
`;

exports[`BaseTaskList Loading State renders correct markup 1`] = `
<div>
  <div role="list" class="v-list v-sheet v-sheet--tile theme--light">
    <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
      <div class="v-skeleton-loader__list-item-avatar v-skeleton-loader__bone">
        <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
        <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      </div>
    </div>
    <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
      <div class="v-skeleton-loader__list-item-avatar v-skeleton-loader__bone">
        <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
        <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      </div>
    </div>
    <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
      <div class="v-skeleton-loader__list-item-avatar v-skeleton-loader__bone">
        <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
        <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      </div>
    </div>
    <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
      <div class="v-skeleton-loader__list-item-avatar v-skeleton-loader__bone">
        <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
        <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      </div>
    </div>
    <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
      <div class="v-skeleton-loader__list-item-avatar v-skeleton-loader__bone">
        <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
        <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      </div>
    </div>
  </div>
  <!---->
  <!---->
</div>
`;

exports[`BaseTaskList With Pinned Tasks renders correct markup 1`] = `
<div>
  <!---->
  <!---->
  <div role="list" class="v-list v-sheet v-sheet--tile theme--light">
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 6 (pinned)</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light primary--text"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 1</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 2</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 3</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 4</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
    <div tabindex="-1" role="listitem" class="v-list-item theme--light">
      <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title">Task 5</div>
      </div>
      <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;
